release.nix

```diff
@@ -0,0 +1,89 @@
+let
+  fetchNixpkgs = import ./fetchNixpkgs.nix;
+
+  nixpkgs = fetchNixpkgs {
+    rev = "c5f9cd4cde81bcef2c8882d359d9d30313bebeb3";
+
+    sha256 = "12w6g50hgk5nmijgs3rv7kq5mdybnkyv3cr2zm0q3ka9ax180dk6";
+
+    outputSha256 = "1nilhz6rhfg3ckp8yfmgy6v3q6spbyxfg0yn8rc2ydyn7119h4fn";
+  };
+
+  overlay = pkgsNew: pkgsOld: {
+    check-source = pkgsNew.writeText "check-source.py" ''
```

Comment:
Can’t we keep this as a file in the repo? It’s going to be a lot easier to maintain that way

Reply:
I'm not sure we can keep it in the repo, as to run the script with CI in nix, we need to patch the executables for `dhall` ([here](https://github.com/dhall-lang/dhall-kubernetes/pull/23/commits/597b56408e9711a71cfa73d08b03292bfef60f70#diff-6ab84d77bf6d73598ebba867195ce313R43)) and `bash` ([here](https://github.com/dhall-lang/dhall-kubernetes/pull/23/commits/597b56408e9711a71cfa73d08b03292bfef60f70#diff-6ab84d77bf6d73598ebba867195ce313R44)) (as nix keeps every version installed in a separate path, and in the job we deterministically specify the version of every dependency - as we pin the revision of the `nixpkgs` repo).

We could do it with `sed`, but I think it's fine (and equally easy to maintain) to keep it in the nix file, as:
- the users are not supposed to run the script, but it's just supposed to run in CI as a smoketest (will run on PR branches and `master`)
- if the users actually want to run it, it's just `nix-build release.nix` (we can add this to the readme I guess)
- my hope is that we could get rid of it entirely: if we go with #7 we won't need it, as we would just build the Dhall AST directly from the Haskell library (so it's correct by construction)